integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+        });
+    }
+
+    @Test
+
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoMySQL.class })
+    public void testJoinInlineWithLimitUnion() {
+        doInJPA(entityManager -> {
+            QRecursiveEntity t = new QRecursiveEntity("t");
+            QRecursiveEntity subT = new QRecursiveEntity("subT");
+            QRecursiveEntity subT2 = new QRecursiveEntity("subT2");
+            QRecursiveEntity subT3 = new QRecursiveEntity("subT3");
+
+            JPQLQuery<RecursiveEntity> subA = select(
+                    new Binds<RecursiveEntity>()
+                        .bind(recursiveEntity.id, subT.id)
+                        .bind(recursiveEntity.name, subT.name)
+                        .bind(recursiveEntity.parent, subT.parent)
+            ).from(subT);
+
+            JPQLQuery<RecursiveEntity> subB = select(
+                    new Binds<RecursiveEntity>()
+                        .bind(recursiveEntity.id, subT3.id)
+                        .bind(recursiveEntity.name, subT3.name)
+                        .bind(recursiveEntity.parent, subT3.parent)).from(subT3);
+
+            SubQueryExpression<RecursiveEntity> union = new BlazeJPAQuery<RecursiveEntity>().unionAll(
+                    subA,
+                    subB
+            );
+
+            new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .select(t, subT2)
+                    .from(t)
+                    .leftJoin(union, subT2).on(subT2.eq(subT2))
+                    .fetch();
+        });
+    }
+
+    @Test
+
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoMySQL.class })
+    public void testJoinInlineEntityWithLimitUnion() {
+        doInJPA(entityManager -> {
+            QRecursiveEntity t = new QRecursiveEntity("t");
+            QRecursiveEntity subT = new QRecursiveEntity("subT");
+            QRecursiveEntity subT2 = new QRecursiveEntity("subT2");
+            QRecursiveEntity subT3 = new QRecursiveEntity("subT3");
+
+            JPQLQuery<RecursiveEntity> subA = selectFrom(subT).where(subT.id.lt(5L));
+            JPQLQuery<RecursiveEntity> subB = selectFrom(subT3).where(subT3.id.gt(10L));
+
+            SubQueryExpression<RecursiveEntity> union = new BlazeJPAQuery<RecursiveEntity>().unionAll(
+                    subA,
+                    subB
+            );
+
+            new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .select(t, subT2)
+                    .from(t)
+                    .leftJoin(union, subT2).on(subT2.eq(subT2))
+                    .fetch();
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoH2.class, NoMySQLOld.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet` and also `// NOTE: Entity joins are only supported on Hibernate 5.1+`
Also add comments for the other exclusions. The Oracle and old MySQL version we test with doesn't support lateral, H2 also doesn't.